### DISCUSSION TITLE
Linuxserver SWAG in combination with NiceGUI

### DISCUSSION THREAD

**Discussion by nijenhuisb** [zauberzeug/nicegui#1112] [Q&A]
### Question

Hi all,

I have a server running multiple Docker services (Grafana, InfluxDB, Telegraf amongst others), accessible from the outside world through SWAG (https://hub.docker.com/r/linuxserver/swag) and I would like to add a NiceGUI app to this. So far so good, the app runs locally and is accessible through the local IP address of the machine. However, when I try to open the app through the web address routed through SWAG, the message "Connection lost. Trying to reconnect..." shows up in the browser.

Steps taken:
- When I shutdown the app, I receive a 502 at the location of the app.
- When restarting the app, the "Connection lost. Trying to reconnect..." message returns.
- This makes me think the SWAG config is not the problem? I use the _template.subfolder.config file from SWAG. Am I right? Anyone has experiences with SWAG + NiceGUI?

#714 Talks about a similar message but it does not seem to be related and looks solved

**Comment by rodja** [comment]
I'm not familiar with SWAG but it seems to use Nginx. Maybe the websocket stuff is not sufficiently configured? Maybe you can compare with our working [nginx example](https://github.com/zauberzeug/nicegui/tree/main/examples/nginx_subpath):

https://github.com/zauberzeug/nicegui/blob/a464337f1861ac2292852cb17a4d1ecf9bada2eb/examples/nginx_subpath/nginx.conf#L47-L53

**Reply by nijenhuisb** [reply]
Thanks! I've tried this indeed (SWAG uses nginx for reverse proxy), setting the headers etc. It still gave me the 'Trying to connect...' message. So somehow, SWAG/nginx routes correctly but then the necessary files cannot be loaded?

I ended up applying a subdomain instead of a subfolder. That works! (So I guess this discussion is still open?)

 So that is all working, however, I'm embedding Grafana iframes now, in Chrome this opens nicely, but in mobile Safari on iOS this is problematic. Any experiences with this?

**Reply by rodja** [reply]
> I'm embedding Grafana iframes now, in Chrome this opens nicely, but in mobile Safari on iOS this is problematic. Any experiences with this?

No, I'm sorry. Does Grafana work as expected without embedding it in an iframe?
